### DISCUSSION TITLE
Better yarn/npm(/lerna) workspace support

### DISCUSSION THREAD

**Discussion by marcodejongh** [renovatebot/renovate#9281] [Request Help]
**What would you like Renovate to be able to do?**

Renovate should have better awareness of mono-repos, and gain the ability to create upgrade PR's per workspace package.
For example if we we have the following repository structure:

```
repo/packages/package1
repo/packages/package2
repo/packages/package3
```
And there's a new version of `date-fns`, we should be able to setup Renovate so that it will create separate PR's for every package, so in the case of this example there will be 3 PR's:

* Upgrade data-fns to 1.2.3 in {package1 PackageName}
* Upgrade data-fns to 1.2.3 in {package2 PackageName}
* Upgrade data-fns to 1.2.3 in {package3 PackageName}

But this behaviour should be configurable by dependency. Because some dependencies do require a cross-repo upgrade, for example when upgrading React we want to have just 1 upgrade PR.

This functionality is useful for big FE mono-repos that manage multiple packages that are published to npm.

**Did you already have any implementation ideas?**

No idea, I assume this requires changes in not just platform but also other places to facilitate the creation of multiple PR's

I'd be interested in building this myself

**Comment by viceice** [comment]
Deps should already be grouped by default, I you like to split it per project, you can use `"additionalBranchPrefix": "{{parentDir}}-" ` in a package rule.

https://docs.renovatebot.com/configuration-options/#additionalbranchprefix

**Reply by marcodejongh** [reply]
@viceice We have a team owners ship assignment system that is all filesystem based. 
In the root of the repo we have a teams.json, which defines the teams. Then in the package.json we have a "team" field.
I'd like to use teamName as `parentDir`, it doesn't make sense to add something specific to what we have to Renovate.

But could you think of a way I could add this to Renovate, that is also useful for other users and therefore could be merged in?

**Reply by viceice** [reply]
I don't see a renovate benefit for this specific use case. Best thing you can do is to update renovate package rules based on your team assignments

eg:
```js
{
  matchPaths: ["path1/", "path3/"],
  additionalBranchPrefix: "team1-",
},
{
  matchPaths: ["path2/", "path5/"],
  additionalBranchPrefix: "team2-",
}.
```

**Reply by rarkins** [reply]
Agreed. Maybe add a custom lint/test to the repo to ensure they stay in sync if it's important. Ie so that nobody updates the teams in package.json without also updating the package rules to match

**Reply by marcodejongh** [reply]
We could use some callback like function or plugin system to make it really flexible?
We can't really structure the repo around this, since it's a really big repo and having to move files around everytime a package moves ownership would be too disruptive.

Maybe we could generate the matchPaths, what if we did:
```
{
  matchPaths: ["packages/editor/editor-core", "packages/editor/editor-common"],
  additionalBranchPrefix: "editor-",
},
{
  matchPaths: ["packages/editor/editor-tables"],
  additionalBranchPrefix: "editor-tables-",
},
```
